README.md

```diff
@@ -1,10 +1,157 @@
+* Build Airflow and providers into a Python environment
+* Package it as an OCI image wrapped in a `.rock` file (e.g., `airflow-rock_3.1.0_amd64.rock`)
+
+## Running and testing the rock
+
+Once built, you can run the rock locally using `docker`:
+
+### Using Docker
+
+```bash
+# Load the rock into your local Docker daemon
+rockcraft.skopeo --insecure-policy copy oci-archive:airflow-rock_3.1.0_amd64.rock docker-daemon:airflow-rock:3.1.0
+# Run the rock
+docker run -it --rm -p 5000:5000 airflow-rock:3.1.0
+```
+
+The container will start Airflow in standalone mode.
+The Airflow web UI will be available at: http://localhost:5000
+
+### Using Pebble inside the container
+
+```bash
+docker exec -it <container_id> pebble services
+docker exec -it <container_id> pebble logs
+```
+
+## Rock services and checks
+
+The rock defines one service, managed by Pebble:
+
+```yaml
+services:
+  airflow:
+    startup: enabled
+    command: /usr/bin/airflow standalone
+```
+
+This runs Airflow’s built-in standalone mode, which launches:
+
+* An API-Server
+* A scheduler
+* A metadata database (SQLite, by default)
+
+A simple liveness check verifies Airflow is running:
+
+```yaml
+checks:
+  airflow-running:
+    exec:
+      command: pgrep airflow
+```
+
+## Parts Explained
+
+The rock is built from several parts:
+
+| Part                       | Purpose                                                                 |
+| -------------------------- | ----------------------------------------------------------------------- |
+| **airflow**                | Builds Apache Airflow from source and installs providers                |
+| **airflow-license**        | Includes upstream Apache Airflow LICENSE                                |
+| **airflow-rock-license**   | Includes the LICENSE for this rock                                      |
+| **env_stage** *(optional)* | Stages a system-wide `/etc/environment` file with environment variables |
+
+### Python constraints
+
+The build uses Airflow’s official constraints file for a certain version of Python, for example:
+
+```yaml
+python-constraints:
+  - https://raw.githubusercontent.com/apache/airflow/constraints-3.1.0/constraints-3.12.txt
+```
+
+This ensures the pinned versions of all dependencies and providers match the desired version of Airflow.
+
+## Provider packages
+
+The rock installs a comprehensive list of Airflow providers, including:
+
+* Amazon (`apache-airflow-providers-amazon`)
+* Google (`apache-airflow-providers-google`)
+* Kubernetes (`apache-airflow-providers-cncf-kubernetes`)
+* PostgreSQL, MySQL, ODBC, Redis, etc.
+
+To customize providers, edit the `python-packages` list in the `airflow` part.
+### List providers installed
 ```bash
-rockcraft pack
+airflow providers list
+```
+
+## Licensing
+
+Two license files are included in the rock image:
+
+* `licenses/LICENSE-airflow` – Apache 2.0 license for upstream Airflow
+* `licenses/LICENSE-airflow-rock` – License for this rock’s source
```

Comment:
might be good to be explicit about the rock license too (e.g. `Apache 2.0 license for this rock's source`)